pkg/secrets/pushtofile/retrieve_secrets_test.go

```diff
@@ -0,0 +1,152 @@
+					{Alias: "username", Id: "ci/openshift/username"},
+					{Alias: "password", Id: "ci/openshift/password"},
+				},
+				SecretsValues: map[string]string{},
+			},
+		},
+		assert: assertGoodResults(SecretGroups{
+			SecretGroup{
+				Label:                  "cache",
+				FilePath:               "./testdata/cache.cfg",
+				FileTemplate:           "",
+				ConjurSecretPathPrefix: "",
+				SecretSpecs: []SecretSpec{
+					{Alias: "api-url", Id: "dev/openshift/api-url"},
+					{Alias: "username", Id: "dev/openshift/username"},
+					{Alias: "password", Id: "dev/openshift/password"},
+				},
+				SecretsValues: map[string]string{
+					"api-url":  string("https://postgres.example.com"),
+					"username": string("admin"),
+					"password": string("open-$e$ame"),
+				},
+			},
+			SecretGroup{
+				Label:                  "db",
+				FilePath:               "./testdata/db.json",
+				FileTemplate:           "",
+				ConjurSecretPathPrefix: "",
+				SecretSpecs: []SecretSpec{
+					{Alias: "api-url", Id: "ci/openshift/api-url"},
+					{Alias: "username", Id: "ci/openshift/username"},
+					{Alias: "password", Id: "ci/openshift/password"},
+				},
+				SecretsValues: map[string]string{
+					"api-url":  string("https://ci.postgres.example.com"),
+					"username": string("administrator"),
+					"password": string("open-$e$ame"),
+				},
+			},
+		}),
+	},
+	{
+		description: "Bad ID",
+		contents: SecretGroups{
+			SecretGroup{
+				Label:                  "cache",
+				FilePath:               "./testdata/cache.cfg",
+				FileTemplate:           "",
+				ConjurSecretPathPrefix: "",
+				SecretSpecs: []SecretSpec{
+					{Alias: "api-url", Id: "foo/openshift/bar"},
+					{Alias: "username", Id: "dev/openshift/username"},
+					{Alias: "password", Id: "dev/openshift/password"},
+				},
+				SecretsValues: map[string]string{},
+			},
+			SecretGroup{
+				Label:                  "db",
+				FileTemplate:           "",
+				FilePath:               "./testdata/db.json",
+				ConjurSecretPathPrefix: "",
+				SecretSpecs: []SecretSpec{
+					{Alias: "api-url", Id: "ci/openshift/api-url"},
+					{Alias: "username", Id: "ci/openshift/username"},
+					{Alias: "password", Id: "ci/openshift/password"},
+				},
+				SecretsValues: map[string]string{},
+			},
+		},
+		assert: func(t *testing.T, result SecretGroups, err error) {
+			assert.Contains(t, err.Error(), "Failed to retrieve secrets")
+		},
+	},
+}
+
+func TestRetrieveSecrets(t *testing.T) {
+	var mockAccessToken mocks.MockAccessToken
+	prepareMockDBs()
+	retrieveConjurSecretsFunc = mocks.RetrieveConjurSecrets
+	var s SecretGroups
+	for _, tc := range retrieveSecretsTestCases {
+		t.Run(tc.description, func(t *testing.T) {
+			s = tc.contents
+			fmt.Printf("TestRetrieveSecrets %t\n", mocks.CanExecuteConjurVar)
+			err := IterateOverGroups(mockAccessToken, &s)
+			tc.assert(t, s, err)
+		})
+	}
+}
+
+func prepareMockDBs() {
+	mocks.CanExecuteConjurVar = true
```

Comment:
Here's another example where the existing Secrets Provider code uses global variables for test mocking. (I chatted with @doodlesbykumbi  about this earlier today.) This means that tests probably can't be run in parallel, since parallel tests may be writing conflicting values to the global vars.

I don't think we should hold up your merge for this, but we'll probably need to file a ticket to eventually fix this.

Reply:
Rebased to [Refactor mocks packages](https://github.com/cyberark/secrets-provider-for-k8s/pull/368)